### ISSUE TITLE
Menu items not working.

### ISSUE THREAD

**Issue by Sential** [qazbnm456/lulumi-browser#12]
When you click on one of the menu items, Electron will popup error. For example when I click on `Preferences`, Electron is throwing me an error.

Screenshot:
![screenshot](http://i.imgur.com/zMCljHw.png)

**Comment by qazbnm456** [comment]
@Sential Hi there, thanks for trying out. From your screenshot, it seems like lulumi cannot locate the `.git` directory. Would you like to provide me more detailed information about your environment and running environment(developing or production)? Thank you.

**Comment by Sential** [comment]
Sure, I built source code of release 1.3.0, because newest version throws me errors when I'm trying to run `yarn run dev`. My operating system is Windows 10.

**Comment by qazbnm456** [comment]
It turns out that the release tarball and zip file do not contain `.git` directory.

When building from the given tarball, lulumi searches for `.git` directory to filled out the `rev` entry, and because `.git` doesn't be there, the error, therefore, is thrown out.
Do a clean clone via `git clone` before proceeding building process should solve this.

Would you try again and report back the outcome, @Sential ?

**Comment by Sential** [comment]
There's another error.

```
$ yarn run lint:fix && cd app && yarn install
yarn run v0.21.3
$ eslint --ext .js,.vue -f ./node_modules/eslint-friendly-formatter --fix app test

  ✘  http://eslint.org/docs/rules/  Parsing error: Adjacent JSX elements must be wrapped in an enclosing tag

  25 | </template>
  26 |
> 27 | <script>
     | ^
  28 |   import store from './vuex/store';
  29 |
  30 |   export default {                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
  C:\Users\Sential\Desktop\lulumi-browser\app\src\guest\renderer\App.vue:33:1
  <script>
   ^

  ✘  http://eslint.org/docs/rules/  Parsing error: Unexpected token, expected ,

  12 |     h2#about-desc(ref="h2")
  13 |     li(v-for="data in datas.about")
> 14 |       router-link(:to="`/${data[1]}`", :key="$route.path") {{ data[0] }}
     |                                                                   ^
  15 | </template>
  16 |
  17 | <script>                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     
  C:\Users\Sential\Desktop\lulumi-browser\app\src\guest\renderer\components\AboutMainView.vue:22:67
        router-link(:to="`/${data[1]}`", :key="$route.path") {{ data[0] }}
                                                                     ^

  ✘  http://eslint.org/docs/rules/  Parsing error: Unexpected token, expected ,

  87 |             el-progress(:status="checkStateForProgress(file.state)", :text-inside="true", :percentage="percentage(file)", :stroke-width="18", class="download-list__item-progress")
  88 |             a(class="download-list__item-name download-list__item-link", :href="`${file.url}`")
> 89 |               i(class="el-icon-document") {{ file.name }}
     |                                                  ^
  90 |             el-button-group
  91 |               el-button(:disabled="file.state !== 'progressing'", v-if="file.isPaused && file.canResume", :plain="true", type="warning", size="small", icon="caret-right", @click="resumeDownload(file.startTime)")
  92 |               el-button(:disabled="file.state !== 'progressing'", v-else, :plain="true", type="warning", size="small", icon="minus", @click="pauseDownload(file.startTime)")
  C:\Users\Sential\Desktop\lulumi-browser\app\src\guest\renderer\components\AboutMainView\Downloads.vue:111:50
                i(class="el-icon-document") {{ file.name }}
                                                    ^

  ✘  http://eslint.org/docs/rules/  Parsing error: Unexpected token

  38 |               | {{ extension }}
  39 |             div
> 40 |               | {{ "Path: " }}
     |                             ^
  41 |               span(class="extensions-list__item-path") {{ loadPath(extension) }}
  42 | </template>
  43 |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              
  C:\Users\Sential\Desktop\lulumi-browser\app\src\guest\renderer\components\AboutMainView\Extensions.vue:57:29
                | {{ "Path: " }}
                               ^

  ✘  http://eslint.org/docs/rules/  Parsing error: Unexpected token

  118 |         h1 History
  119 |       el-col(:span="4")
> 120 |         div {{ `History is ${sync === true ? 'syncing' : 'not syncing'}...&nbsp;` }}
      |                ^
  121 |         el-switch(@change="toggleSync", v-model="sync", on-color="#13ce66", off-color="#ff4949")
  122 |       el-col(:span="12")
  123 |         el-input#history-filter(@focus="toggleSync(false)", @blur="toggleSync(true)", placeholder="Input keyword", v-model="filterText")                                                                                                                                                                                                                                                                                                                                      
  C:\Users\Sential\Desktop\lulumi-browser\app\src\guest\renderer\components\AboutMainView\History.vue:137:16
          div {{ `History is ${sync === true ? 'syncing' : 'not syncing'}...&nbsp;` }}
                  ^

  ✘  http://eslint.org/docs/rules/  Parsing error: Unexpected token, expected ,

  14 |       el-table-column(label="version", width="180", align="center")
  15 |         template(scope="scope")
> 16 |           a.cell(v-if="scope.row.key === 'rev'", :href="`https://github.com/qazbnm456/lulumi-browser/commit/${scope.row.value}`") {{ scope.row.value.substring(0, 7) }}
     |                                                                                                                                           ^
  17 |           .cell(v-else) {{ scope.row.value }}
  18 | </template>
  19 |                                                                                                                                                                                                                                                                                    
  C:\Users\Sential\Desktop\lulumi-browser\app\src\guest\renderer\components\AboutMainView\Lulumi.vue:27:139
            a.cell(v-if="scope.row.key === 'rev'", :href="`https://github.com/qazbnm456/lulumi-browser/commit/${scope.row.value}`") {{ scope.row.value.substring(0, 7) }}
                                                                                                                                             ^

  ✘  http://eslint.org/docs/rules/  Parsing error: Unexpected token, expected ,

  13 |       el-menu#menu(theme="dark", class="el-menu-vertical-demo", :router="true")
  14 |         div(v-for="(data, index) in datas.preferences", :key="index")
> 15 |           el-menu-item(:index="`${path}/${data[1]}`") {{ data[0] }}
     |                                                              ^
  16 |     el-col(:span="18", :offset="1")
  17 |       router-view
  18 | </template>                                                                                                                                                                                                                                                                                                                                                                      
  C:\Users\Sential\Desktop\lulumi-browser\app\src\guest\renderer\components\AboutMainView\Preferences.vue:25:62
            el-menu-item(:index="`${path}/${data[1]}`") {{ data[0] }}
                                                                ^

  ✘  http://eslint.org/docs/rules/  Parsing error: Adjacent JSX elements must be wrapped in an enclosing tag

  41 | </template>
  42 |
> 43 | <script>
     | ^
  44 |   export default {
  45 |     data() {
  46 |       return {                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
  C:\Users\Sential\Desktop\lulumi-browser\app\src\guest\renderer\components\AboutMainView\Preferences\Homepage.vue:55:1
  <script>
   ^

  ✘  http://eslint.org/docs/rules/  Parsing error: Adjacent JSX elements must be wrapped in an enclosing tag

  49 | </template>
  50 |
> 51 | <script>
     | ^
  52 |   export default {
  53 |     data() {
  54 |       return {                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
  C:\Users\Sential\Desktop\lulumi-browser\app\src\guest\renderer\components\AboutMainView\Preferences\PDFViewer.vue:61:1
  <script>
   ^

  ✘  http://eslint.org/docs/rules/  Parsing error: Unexpected token, expected }

  50 |   div
  51 |     h1 Search Engine Provider Page
> 52 |     el-table(:data="tableData", :default-sort = "{prop: 'name', order: 'descending'}", :row-class-name="currentSearchEngine", @current-change="handleCurrentChange")
     |                                                       ^
  53 |       el-table-column(prop="current", label="Current", width="180", align="center")
  54 |       el-table-column(prop="search", label="Search Engine", align="center")
  55 |       el-table-column(prop="name", label="Name", width="180", align="center")                                                                                                                                                                                                                                                                                                    
  C:\Users\Sential\Desktop\lulumi-browser\app\src\guest\renderer\components\AboutMainView\Preferences\SearchEngineProvider.vue:61:55
      el-table(:data="tableData", :default-sort = "{prop: 'name', order: 'descending'}", :row-class-name="currentSearchEngine", @current-change="handleCurrentChange")
                                                         ^

  ✘  http://eslint.org/docs/rules/  Parsing error: Adjacent JSX elements must be wrapped in an enclosing tag

  55 | </template>
  56 |
> 57 | <script>
     | ^
  58 |   export default {
  59 |     data() {
  60 |       return {                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
  C:\Users\Sential\Desktop\lulumi-browser\app\src\guest\renderer\components\AboutMainView\Preferences\TabConfig.vue:80:1
  <script>
   ^

  ✘  http://eslint.org/docs/rules/  Parsing error: Adjacent JSX elements must be wrapped in an enclosing tag

  11 | </template>
  12 |
> 13 | <script>
     | ^
  14 |   import store from 'renderer/vuex/store';
  15 |
  16 |   export default {                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         
  C:\Users\Sential\Desktop\lulumi-browser\app\src\renderer\App.vue:19:1
  <script>
   ^

  ✘  http://eslint.org/docs/rules/  Parsing error: Unexpected token, expected ,

  979 |     #footer
  980 |       transition(name="extend")
> 981 |         .browser-page-status(v-show="page.statusText") {{ page.statusText }}
      |                                                               ^
  982 |       download(v-show="$store.getters.downloads.length !== 0 && showDownloadBar")
  983 | </template>
  984 |                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
  C:\Users\Sential\Desktop\lulumi-browser\app\src\renderer\components\BrowserMainView.vue:994:63
          .browser-page-status(v-show="page.statusText") {{ page.statusText }}
                                                                 ^

  ✘  http://eslint.org/docs/rules/  Parsing error: Unexpected token, expected ,

  91 |         el-popover(placement="top-start", width="178", trigger="hover", popper-class="download-list__popper"
  92 |           v-for="(file, index) in files", :key="index")
> 93 |           div {{ file.state }}
     |                      ^
  94 |           el-button-group(v-show="checkStateForButtonGroup(file.state)")
  95 |             el-button(:disabled="file.state !== 'progressing'", v-if="file.isPaused && file.canResume", :plain="true", type="warning", size="mini", icon="caret-right", @click="resumeDownload(file.startTime)")
  96 |             el-button(:disabled="file.state !== 'progressing'", v-else, :plain="true", type="warning", size="mini", icon="minus", @click="pauseDownload(file.startTime)")                                                                                                          
  C:\Users\Sential\Desktop\lulumi-browser\app\src\renderer\components\BrowserMainView\Download.vue:116:22
            div {{ file.state }}
                        ^

  ✘  http://eslint.org/docs/rules/  Parsing error: Adjacent JSX elements must be wrapped in an enclosing tag

  267 | </template>
  268 |
> 269 | <script>
      | ^
  270 |   import Vue from 'vue';
  271 |   import url from 'url';
  272 |   import { focus } from 'vue-focus';                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      
  C:\Users\Sential\Desktop\lulumi-browser\app\src\renderer\components\BrowserMainView\Navbar.vue:304:1
  <script>
   ^

  ✘  http://eslint.org/docs/rules/  Parsing error: Unexpected token

  151 | <template lang="pug">
  152 |   el-row(:gutter="20", type="flex", align="middle", justify="space-between", style="width: 100vw; flex-direction: row;")
> 153 |     el-col(:span="18") {{ `${template}` }}
      |                           ^
  154 |     el-col(:span="6")
  155 |       el-button(:plain="true", type="success", size="small", @click="onAllow") Allow
  156 |       el-button(:plain="true", type="danger", size="small", @click="onDeny") Deny                                                                                                                                                                                                                                                                                                                                                                                             
  C:\Users\Sential\Desktop\lulumi-browser\app\src\renderer\components\BrowserMainView\Notification.vue:161:27
      el-col(:span="18") {{ `${template}` }}
                             ^

  ✘  http://eslint.org/docs/rules/  Parsing error: Adjacent JSX elements must be wrapped in an enclosing tag

  210 | </template>
  211 |
> 212 | <script>
      | ^
  213 |   import urlUtil from 'renderer/js/lib/url-util';
  214 |
  215 |   import Event from 'src/api/extensions/event';                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           
  C:\Users\Sential\Desktop\lulumi-browser\app\src\renderer\components\BrowserMainView\Page.vue:227:1
  <script>
   ^

  ✘  http://eslint.org/docs/rules/  Parsing error: Adjacent JSX elements must be wrapped in an enclosing tag

  24 | </template>
  25 |
> 26 | <script>
     | ^
  27 |   import Icon from 'vue-awesome/components/Icon';
  28 |   import 'vue-awesome/icons/arrow-left';
  29 |   import 'vue-awesome/icons/arrow-right';                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
  C:\Users\Sential\Desktop\lulumi-browser\app\src\renderer\components\BrowserMainView\SwipeArrow.vue:34:1
  <script>
   ^

  ✘  http://eslint.org/docs/rules/  Parsing error: Unexpected token, expected ,

  104 |           el-tooltip(:content="page.title || 'loading'", placement="bottom", :openDelay="1000")
  105 |             span(class="chrome-tab-title")
> 106 |               | {{ page.title || 'loading' }}
      |                        ^
  107 |         a.close(@click.stop="$parent.onTabClose(index)", class="chrome-tab-close")
  108 |         svg(width="15", height="30", class="right-edge")
  109 |           path(class="edge-bg", d="m14,29l0,-28l-2,0.1l-11.45,27.9l13.2,0z", stroke-linecap="null", stroke-linejoin="null", stroke-dasharray="null", stroke-width="0")                                                                                                                                                                                                          
  C:\Users\Sential\Desktop\lulumi-browser\app\src\renderer\components\BrowserMainView\Tabs.vue:136:24
                | {{ page.title || 'loading' }}
                          ^


✘ 19 problems (19 errors, 0 warnings)


Errors:
  19  http://eslint.org/docs/rules/null
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```
Screenshot:
[screen](http://i.imgur.com/adkHPbl.png)

**Comment by qazbnm456** [comment]
@Sential Okay, I have encountered this situation before, and I have no clue what wrong it's, because everything works just fine on macOS and Linux. It only happened on Windows platform.
Maybe it's a platform related issue, and I will check this out recently.

Still thank you for your testing. I've uploaded Windows version one [here](https://github.com/qazbnm456/lulumi-browser/releases/download/v1.3.0/lulumi-browser-win32-x64.zip), and it's built with wine on Ubuntu.
